modules/machineset-azure-capacity-reservation.adoc

```diff
@@ -0,0 +1,102 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating_machinesets/creating-machineset-azure.adoc
+// * machine_management/control_plane_machine_management/cpmso_provider_configurations/cpmso-config-options-azure.adoc
+
+ifeval::["{context}" == "cpmso-using"]
+:cpmso:
+endif::[]
+
+:_mod-docs-content-type: PROCEDURE
+[id="machineset-azure-capacity-reservation_{context}"]
+= Configuring Capacity Reservation by using machine sets
+
+//4.17 feature backported to 4.16.3, 4.15.25, and 4.14.35.
+{product-title} version 4.14.35 and later supports on-demand Capacity Reservation with Capacity Reservation groups on {azure-full} clusters.
+
+You can configure a machine set to deploy machines on any available resources that match the parameters of a capacity request that you define.
+These parameters specify the VM size, region, and number of instances that you want to reserve.
+If your {azure-short} subscription quota can accommodate the capacity request, the deployment succeeds.
+
+For more information, including limitations and suggested use cases for this {azure-short} instance type, see the {azure-full} documentation about link:https://learn.microsoft.com/en-us/azure/virtual-machines/capacity-reservation-overview[On-demand Capacity Reservation].
```

Comment:
This PR is essentially a cherry pick, so I will not make this change here. I think it would be valid to update the main version to the following:
```suggestion
For more information, including limitations and suggested use cases for this {azure-short} instance type, see link:https://learn.microsoft.com/en-us/azure/virtual-machines/capacity-reservation-overview[On-demand Capacity Reservation] ({azure-full} documentation).
```